### PR TITLE
UPDATE: modifying 'for ... of' to 'for ... in' and array destructurin…

### PR DESCRIPTION
**Pull request by msuperina**
…g syntaxes to prevent using Symbol in final build

### CODE REVIEW

File: dist/webstomp.js [hunk 1]
```diff
@@ -74,7 +74,7 @@ return /******/ (function(modules) { // webpackBootstrap
 	    // This method creates a WebSocket client that is connected to
 	    // the STOMP server located at the url.
 	    client: function client(url) {
-	        var options = arguments.length <= 1 || arguments[1] === undefined ? { protocols: _utils.VERSIONS.supportedProtocols() } : arguments[1];
+	        var options = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : { protocols: _utils.VERSIONS.supportedProtocols() };
```
Comment (msuperina) [inline dist/webstomp.js:77]: it appears that the build is generating new logic for conditional tests. I am adding below comments on the lines I actually changed

File: src/client.js [hunk 2]
```diff
@@ -62,7 +62,10 @@ class Client {
     // The errorCallback is optional and the 2 first forms allow to pass other
     // headers in addition to `client`, `passcode` and `host`.
     connect(...args) {
-        let [headers, connectCallback, errorCallback] = this._parseConnect(...args);
+        let parseConnect = this._parseConnect(...args);
```
Comment (msuperina) [inline src/client.js:65]: Instead of array destructuring, explicitely defining variables.

File: src/client.js [hunk 3]
```diff
@@ -332,7 +335,9 @@ class Client {
         // heart-beat header received from the server looks like:
         //
         //     heart-beat: sx, sy
-        const [serverOutgoing, serverIncoming] = (headers['heart-beat'] || '0,0').split(',').map(v => parseInt(v, 10));
+        const heartBeatHeaders = (headers['heart-beat'] || '0,0').split(',').map(v => parseInt(v, 10));
```
Comment (msuperina) [inline src/client.js:338]: Instead of array destructuring, explicitely defining variables.

File: src/frame.js [hunk 4]
```diff
@@ -46,7 +46,7 @@ class Frame {
 
         // Parse headers in reverse order so that for repeated headers, the 1st
         // value is used
-        for (let line of headerLines.reverse()) {
+        for (let line in headerLines.reverse()) {
```
Comment (msuperina) [inline src/frame.js:49]: seems equivalent in this case or am I missing something ?

**Comment by msuperina** [comment]
The reason for this pull request is because we are using this library with Typescript. 
When using in Internet Explorer, even with Symbol polyfills, runtime errors occur when trying to evaluate System.iterator that is generated with random method names.
This fix solves the problem and removes dependency on Symbol usage.
I wish there were tests to prove nothing is broken.

**Comment by JSteunou** [comment]
I'm sorry but that's really not the way to fix your issue. The src are in standard ES6, if there is any issue with the dist files, the fix should be around the babel project or an update of babel version here and a rebuild or maybe a transcript from ES6 to TypeScript.

**Comment by msuperina** [comment]
Hi thanks for your answer.
I was expecting this outcome.
Please just keep in mind that by using es6 syntax no matter what you force
final users to download polyfills that at the end consume a lot of bandwith.

**Comment by JSteunou** [comment]
You are not forced to use the ES6 files on the client side. dist files are compatible with most of people browsers and by adding babel-polyfill for others it works. It is quite a small price to be ready for using ES6 and more.

In my apps I just add this webpack style loc

``` js
// Check on polyfill
function checkES2015(callback) {
    if ('Symbol' in window) {
        callback();
    } else {
        require.ensure(['babel-polyfill'], function(require) {
            require('babel-polyfill');
            callback();
        }, 'babel-polyfill');
    }
}
```

So polyfills are built a part and only load on old browsers.

File: src/frame.js (hunk 4 above)
Comment (JSteunou) [reply src/frame.js:49]: You actually miss the differences. TLDR; for of is safer & more powerful https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in

Comment (msuperina) [reply src/frame.js:49]: sorry I feel very stupid for this one

**Pull request closed without merging**